src/charm.py

```diff
@@ -564,12 +543,12 @@ def _on_secret_changed(self, event):
         for backup tool on non-leader units to keep them working with MongoDB. The same workflow
         occurs on TLS certs change.
         """
-        if self._compare_secret_ids(
-            event.secret.id, self.app_peer_data.get(Config.Secrets.SECRET_INTERNAL_LABEL)
+        if event.secret.get_info().id == self.app_peer_data.get(
+            Config.Secrets.SECRET_INTERNAL_LABEL
         ):
             scope = APP_SCOPE
-        elif self._compare_secret_ids(
-            event.secret.id, self.unit_peer_data.get(Config.Secrets.SECRET_INTERNAL_LABEL)
+        elif event.secret.get_info().id == self.unit_peer_data.get(
+            Config.Secrets.SECRET_INTERNAL_LABEL
```

Comment:
I think we should use the URI here... ID may overlap across charms, while URI (that's BTW including the ID) never would. What do U think?